### PR TITLE
OCS label nodes post node expansion

### PR DESCRIPTION
**Pull request by shreekarSS**

### CODE REVIEW

**Comment by shreekarSS** [comment]
pr applies only for AWS IPI

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py [hunk 1]
```diff
@@ -56,6 +58,19 @@ def test_add_node(self):
                 node_names=helpers.get_worker_nodes(),
                 status=constants.NODE_READY
             )
+            all_nodes = helpers.get_worker_nodes()
+
+            new_spun_nodes = list(
```
Comment (hnallurv) [inline tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:63]: suggest changing the name of this variable to 'newly_added_nodes'

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py [hunk 2]
```diff
@@ -56,6 +58,19 @@ def test_add_node(self):
...
+                set(all_nodes) - set(initial_nodes)
+            )
+            logging.info('labeling:')
+            node_obj = ocp.OCP(kind='node')
+            for new_node in new_spun_nodes:
+                node_obj.add_label(
+                    resource_name=new_node,
+                    label=constants.OPERATOR_NODE_LABEL
+                )
```
Comment (hnallurv) [inline tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:72]: Please add a check whether the label was successfully added to the nodes or not

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py [hunk 3]
```diff
@@ -56,6 +58,19 @@ def test_add_node(self):
...
             )
+            all_nodes = helpers.get_worker_nodes()
+
+            new_spun_nodes = list(
+                set(all_nodes) - set(initial_nodes)
+            )
+            logging.info('labeling:')
+            node_obj = ocp.OCP(kind='node')
+            for new_node in new_spun_nodes:
+                node_obj.add_label(
```
Comment (ramkiperiy) [inline tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:69]: Why can't we use this function t add node and to label the node https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/node.py#L242 ?
This function needs update to get the label as param from user.

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py [hunk 4]
```diff
@@ -25,6 +25,8 @@ def test_add_node(self):
         """
         dt = config.ENV_DATA['deployment_type']
         if dt == 'ipi':
```
Comment (ramkiperiy) [inline tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:27]: Any plan in future to use this TC for other platform node add too ? if so will be good to have the params parametrized

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py (hunk 3 above)
Comment (shreekarSS) [reply tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:69]: got it , will use add_new_node_and_label_it()

**Comment by shreekarSS** [comment]
===============15:27:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-machine-api --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig get machinesets sshreeka-awsone-sj9mk-worker-us-east-2a -n openshift-machine-api -o yaml
15:27:59 - MainThread - ocs_ci.ocs.machine - INFO - New spun node reached Ready state
15:27:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig get Node  --selector=node-role.kubernetes.io/worker -o yaml
15:27:59 - MainThread - ocs_ci.ocs.node - INFO - New spun node is ['ip-10-0-137-114.us-east-2.compute.internal']
15:27:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig label node ip-10-0-137-114.us-east-2.compute.internal cluster.ocs.openshift.io/openshift-storage=''
15:27:59 - MainThread - ocs_ci.ocs.node - INFO - Successfully labeled ['ip-10-0-137-114.us-east-2.compute.internal'] with OCS storage label
15:27:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig get Node  --selector=node-role.kubernetes.io/worker -o yaml
15:28:00 - MainThread - tests.manage.z_cluster.cluster_expansion.test_node_expansion - INFO - The worker nodes number after 4
PASSED                                                                                                                                                                [100%]
----------------------------------------------------------------------------- live log teardown -----------------------------------------------------------------------------
15:28:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
15:28:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
15:28:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-57fd596bb4-zf4zn ceph health
15:28:01 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
15:28:01 - MainThread - tests.conftest - INFO - Ceph health check passed at teardown


============================================================================= warnings summary ==============================================================================
venv/lib/python3.6/site-packages/_pytest/config/__init__.py:891
  /home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py:891: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: pytest_reportportal
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================================= 1 passed, 1 warning in 263.46s (0:04:23) =================================================================

File: tests/manage/z_cluster/cluster_expansion/test_node_expansion.py (hunk 4 above)
Comment (shreekarSS) [reply tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:25]: on the way ,in my next pr

**Pull request merged**